src/backend/language.cpp

```diff
@@ -79,238 +80,1049 @@ LanguageMap::LanguageMap() {
     //  addLanguage({"rm"}, QT_TRANSLATE_NOOP("QObject", "Raeto-Romance"));
+    addLanguage({"rmc"}, QT_TRANSLATE_NOOP("QObject", "Carpathian Romani"));
+    addLanguage({"rmq"}, QT_TRANSLATE_NOOP("QObject", "Caló"));
+    addLanguage({"rmy"}, QT_TRANSLATE_NOOP("QObject", "Vlax Romani"));
     //  addLanguage({"rn"}, QT_TRANSLATE_NOOP("QObject", "Rundi"));
     addLanguage({"ro"}, QT_TRANSLATE_NOOP("QObject", "Romanian"));
+    addLanguage({"roo"}, QT_TRANSLATE_NOOP("QObject", "Rotokas"));
+    addLanguage({"rop"}, QT_TRANSLATE_NOOP("QObject", "Kriol"));
+    addLanguage({"row"}, QT_TRANSLATE_NOOP("QObject", "Dela-Oenale"));
+    addLanguage({"rro"}, QT_TRANSLATE_NOOP("QObject", "Waima"));
     addLanguage({"ru"}, QT_TRANSLATE_NOOP("QObject", "Russian"));
+    addLanguage({"ruf"}, QT_TRANSLATE_NOOP("QObject", "Luguru"));
+    addLanguage({"rug"}, QT_TRANSLATE_NOOP("QObject", "Roviana"));
     //  addLanguage({"rw"}, QT_TRANSLATE_NOOP("QObject", "Kinyarwanda"));
-    //  addLanguage({"sa"}, QT_TRANSLATE_NOOP("QObject", "Sanskrit"));
+    addLanguage({"rwo"}, QT_TRANSLATE_NOOP("QObject", "Rawa"));
+
+    addLanguage({"sa"}, QT_TRANSLATE_NOOP("QObject", "Sanskrit"));
+    addLanguage({"sab"}, QT_TRANSLATE_NOOP("QObject", "Buglere"));
+    addLanguage({"sah-Cyrl"}, QT_TRANSLATE_NOOP("QObject", "Yakut (Cyrillic script)"));
+    addLanguage({"sbe"}, QT_TRANSLATE_NOOP("QObject", "Saliba"));
     //  addLanguage({"sc"}, QT_TRANSLATE_NOOP("QObject", "Sardinian"));
     addLanguage({"sco"}, QT_TRANSLATE_NOOP("QObject", "Scots"));
     //  addLanguage({"sd"}, QT_TRANSLATE_NOOP("QObject", "Sindhi"));
     //  addLanguage({"se"}, QT_TRANSLATE_NOOP("QObject", "Northern Sami"));
+    addLanguage({"seh"}, QT_TRANSLATE_NOOP("QObject", "Sena"));
+    addLanguage({"sey"}, QT_TRANSLATE_NOOP("QObject", "Secoya"));
     //  addLanguage({"sg"}, QT_TRANSLATE_NOOP("QObject", "Sango"));
+    addLanguage({"sgb"}, QT_TRANSLATE_NOOP("QObject", "Mag-Anchi Ayta"));
+    addLanguage({"sgz"}, QT_TRANSLATE_NOOP("QObject", "Sursurunga"));
+    addLanguage({"shj"}, QT_TRANSLATE_NOOP("QObject", "Shatt"));
+    addLanguage({"shp"}, QT_TRANSLATE_NOOP("QObject", "Shipibo-Conibo"));
     //  addLanguage({"si"}, QT_TRANSLATE_NOOP("QObject", "Sinhalese"));
+    addLanguage({"sim"}, QT_TRANSLATE_NOOP("QObject", "Mende (Papua New Guinea)"));
+    addLanguage({"sja"}, QT_TRANSLATE_NOOP("QObject", "Epena"));
     addLanguage({"sk"}, QT_TRANSLATE_NOOP("QObject", "Slovak"));
     addLanguage({"sl"}, QT_TRANSLATE_NOOP("QObject", "Slovenian"));
+    addLanguage({"sll"}, QT_TRANSLATE_NOOP("QObject", "Salt-Yui"));
     //  addLanguage({"sm"}, QT_TRANSLATE_NOOP("QObject", "Samoan"));
-    //  addLanguage({"sn"}, QT_TRANSLATE_NOOP("QObject", "Shona"));
+    addLanguage({"smk"}, QT_TRANSLATE_NOOP("QObject", "Bolinao"));
+    addLanguage({"sml"}, QT_TRANSLATE_NOOP("QObject", "Central Sama"));
+    addLanguage({"sn"}, QT_TRANSLATE_NOOP("QObject", "Shishona"));
```

Comment:
This should also have the "sna" abbreviation, and be called "Shona" according to https://iso639-3.sil.org/code/sna